modin/pandas/groupby.py

```diff
@@ -120,12 +124,17 @@ def _index_grouped(self):
                 # aware.
                 ErrorMessage.catch_bugs_and_request_email(self._axis == 1)
                 ErrorMessage.default_to_pandas("Groupby with multiple columns")
-                self._index_grouped_cache = {
-                    k: v.index
-                    for k, v in self._df._query_compiler.getitem_column_array(by)
-                    .to_pandas()
-                    .groupby(by=by)
-                }
+                if isinstance(by, list) and all(isinstance(o, str) for o in by):
+                    self._index_grouped_cache = {
+                        k: v.index
+                        for k, v in self._df._query_compiler.getitem_column_array(by)
+                        .to_pandas()
+                        .groupby(by=by)
+                    }
+                else:
+                    by = [o._to_pandas() if isinstance(o, Series) else o for o in by]
+                    pandas_df = self._df._to_pandas().groupby(by=by)
+                    self._index_grouped_cache = {k: v.index for k, v in pandas_df}
```

Comment:
couldn't we get the same dict by using `pandas_df.groups`? That would be more explicit than iterating through groups and creating a dict from it

Reply:
I will look into it.

Reply:
Done.